dashboard/src/modules/components/ComparisonComponent/index.jsx

```diff
@@ -28,16 +28,24 @@ const ComparisonComponent = () => {
   const navigate = useNavigate();
 
   const { datasets } = useSelector((state) => state.overview);
-  const { isCompareSwitchChecked, selectedResourceIds } = useSelector(
-    (state) => state.comparison
-  );
+  const {
+    isCompareSwitchChecked,
+    selectedResourceIds,
+    compareChartData,
+    chartData,
+  } = useSelector((state) => state.comparison);
   useEffect(() => {
-    if (datasets && datasets.length > 0) {
+    if (
+      datasets &&
+      datasets.length > 0 &&
+      !!!compareChartData.length &&
+      !!!chartData.length
+    ) {
       dispatch(getQuisbyData(datasets[0]));
     } else {
       dispatch(getDatasets());
     }
-  }, [datasets, dispatch, navigate]);
+  }, [chartData.length, compareChartData.length, datasets, dispatch, navigate]);
```

Comment:
You're including `navigate` in the closure, but I don't see it referenced in the function.  Should it be removed from this list?